File: Solutions/6kyu/6kyu_the_hunger_games_foxes_and_chickens.js
var caged = new RegExp(/\[([CF\.]+)\]/, 'g')
var cage_remover = new RegExp(/\[[CF\.]+\]/, 'g')
var outsiders = new RegExp(/(^[CF\.]+\[)|(\][CF\.]+\[)|(\][CF\.]+$)/, 'g')


var hungryFoxes = function(farm) {

   // caged chicken
   farm = farm.replace(caged, s => s.includes('F') ? 
                                               s.replace(/C/g, '.')
                                               :
                                               s)
   // outside cage chicken
   if (farm.replace(cage_remover, '').includes('F')) {
     farm = farm.replace(outsiders, s => s.replace(/C/g, '.'))
   }
   
   // no cages and fox outside
   var check = new Set(farm)
   return check.has('F') && !check.has('[') ? farm.replace(/C/g, '.') : farm
 }

File: Solutions/6kyu/6kyu_the_hunger_games_foxes_and_chickens.py
import re

def hungry_foxes(farm):
    
    # caged chicken
    farm = re.sub('\[([CF\.]+)\]', 
                  lambda w: w.group().replace('C','.') if 'F' in w.group() else w.group(),
                     farm)
                     
    # outside cage chicken                 
    if 'F' in re.sub('\[[CF\.]+\]','',farm):
      farm = re.sub('(^[CF\.]+\[)|(\][CF\.]+\[)|(\][CF\.]+$)', 
                  lambda w: w.group().replace('C','.'),
                     farm)
    # if no cages
    check = set(farm)
    return farm.replace('C','.')  if '[' not in check and 'F' in check else farm
